sdk/python/feast/feature_store.py

```diff
@@ -1058,6 +1081,22 @@ def get_online_features(
             features=features, allow_cache=True, hide_dummy_entity=False
         )
 
+        entity_name_to_join_key_map, entity_type_map = self._get_entity_maps(
+            requested_feature_views
+        )
+
+        if native_entity_values:
+            # Convert values to Protobuf once.
+            entity_proto_values: Dict[str, List[Value]] = {
+                k: python_values_to_proto_values(
+                    v, entity_type_map.get(k, ValueType.UNKNOWN)
+                )
+                for k, v in entity_values.items()
+            }
+        else:
+            entity_proto_values = entity_values
+
+        num_rows = _validate_entity_values(entity_proto_values)
```

Comment:
From patching this PR in and doing some changes (https://github.com/feast-dev/feast/pull/2217) looks like the root cause is here. Basically the entity values are `RepeatedValues` (i.e. a list of entity ids for batch retrieval) and those don't work with len() which is used in `_validate_entity_values`

Reply:
I think I've handled this now - though I wonder whether the conversion to a `List` should be done in the FeatureServer on in the FeatureStore method?

Reply:
Yeah especially since there's no other real reason to have them in a proto format. Probably ok for now though